src/KK.AspNetCore.EasyAuthAuthentication/Services/EasyAuthForApplicationsService.cs

```diff
@@ -52,22 +54,28 @@ public AuthenticateResult AuthUser(HttpContext context, ProviderOptions options
         /// <inheritdoc/>
         public bool CanHandleAuthentification(HttpContext httpContext) =>
             IsHeaderSet(httpContext.Request.Headers, AuthorizationHeader) &&
-            httpContext.Request.Headers[AuthorizationHeader].FirstOrDefault().Contains(JWTIdentifyer);
+            httpContext.Request.Headers[AuthorizationHeader].FirstOrDefault().Contains(JWTIdentifier);
 
         private IEnumerable<AADClaimsModel> BuildFromApplicationAuth(JObject xMsClientPrincipal, ProviderOptions options)
         {
-            this.logger.LogDebug($"payload was {xMsClientPrincipal["roles"].ToString()}");
-            var claims = JsonConvert.DeserializeObject<IEnumerable<string>>(xMsClientPrincipal["roles"].ToString())
-                    .Select(r => new AADClaimsModel() { Typ = "roles", Values = r })
+            this.logger.LogDebug($"payload was {xMsClientPrincipal[this.defaultOptions.RoleClaimType].ToString()}");
+
+            var claims = JsonConvert.DeserializeObject<IEnumerable<string>>(xMsClientPrincipal[this.defaultOptions.RoleClaimType].ToString())
```

Comment:
Does this also work when groups in token are injected like:
`{
				"typ": "roles",
				"val": "Sol_Rn3_Bearbeiter"
			},
			{
				"typ": "roles",
				"val": "Sol_Rn91_Developer"
			},
			{
				"typ": "roles",
				"val": "Sol_Rn1_Anforderer"
			},`

Reply:
yes this is working. But here is maybe a little bit of documentation missing, how you can define a custom property to get the roles.